### PR TITLE
Added Project Euler Problem #28 in C++ and Python

### PR DESCRIPTION
**Pull request by arnavb**
**Fixes issue:** N/A

**Changes:**
- Added README for Problem #28
- Added Problem #28 for Project Euler in C++/Python

@AdiChat @iattempt

### CODE REVIEW

File: code/online_challenges/src/project_euler/problem_028/problem_028.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            (i % 2 == 1) &&      // If the current number is odd
+            (isPerfectSquare(i)) // and it is a perfect square
+        )                        // then we have reached the next spiral
+        {
+            incrementRate += 2;
+        }
+    }
+
+    std::cout << diagonalNumberSum << "\n";
+}
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_028/problem_028.cpp]: Forgot newline.

File: code/online_challenges/src/project_euler/problem_028/problem_028.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        diagonal_number_sum += i
+        if i % 2 == 1 and is_perfect_square(i):
+            increment_rate += 2
+
+        i += increment_rate
+    
+    print(diagonal_number_sum)
+
+if __name__ == "__main__":
+    main()
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_028/problem_028.py]: As above.

File: code/online_challenges/src/project_euler/problem_028/problem_028.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    for (int i = 1; i <= limit; i += incrementRate) // Iterate over every diagonal number
+    {
+        diagonalNumberSum += i; // Add the current diagonal number
+        if (
+            (i % 2 == 1) &&      // If the current number is odd
+            (isPerfectSquare(i)) // and it is a perfect square
+        )                        // then we have reached the next spiral
+        {
+            incrementRate += 2;
+        }
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_028/problem_028.cpp]: The braces is not needed here.

File: code/online_challenges/src/project_euler/problem_028/problem_028.cpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{
+    int limit = 1001 * 1001;
+    int incrementRate = 0;
+    int diagonalNumberSum = 0;
+    for (int i = 1; i <= limit; i += incrementRate) // Iterate over every diagonal number
+    {
+        diagonalNumberSum += i; // Add the current diagonal number
+        if (
+            (i % 2 == 1) &&      // If the current number is odd
+            (isPerfectSquare(i)) // and it is a perfect square
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_028/problem_028.cpp]: 1. The newlines of parentheses of `if` is not needed.
2. Put logical operator in front of the conditions, Is it clearer? (If you agree, I'll add to the coding style.)

```
if ((i % 2 == 1)
    && isPerfectSquare(i))
```

Comment (arnavb) [reply code/online_challenges/src/project_euler/problem_028/problem_028.cpp]: Ok, I think it should be added.

File: code/online_challenges/src/project_euler/problem_028/problem_028.cpp (hunk 3 above)
Comment (arnavb) [reply code/online_challenges/src/project_euler/problem_028/problem_028.cpp]: Ok

File: code/online_challenges/src/project_euler/problem_028/problem_028.cpp (hunk 1 above)
Comment (arnavb) [reply code/online_challenges/src/project_euler/problem_028/problem_028.cpp]: At the EOF?

File: code/online_challenges/src/project_euler/problem_028/problem_028.py (hunk 2 above)
Comment (arnavb) [reply code/online_challenges/src/project_euler/problem_028/problem_028.py]: Ok

**Comment by arnavb** [comment]
@iattempt Made changes, please review.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**